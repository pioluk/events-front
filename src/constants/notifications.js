export const DISPLAY_NOTIFICATION = 'DISPLAY_NOTIFICATION'
