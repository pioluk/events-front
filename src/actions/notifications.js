import { DISPLAY_NOTIFICATION } from '../constants/notifications'

export function displayNotification (message) {
  return {
    type: DISPLAY_NOTIFICATION,
    message
  }
}
