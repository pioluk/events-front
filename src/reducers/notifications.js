// @flow

import { DISPLAY_NOTIFICATION } from '../constants/notifications'

type Notification = {
  id: string,
  message: string
}

const initialState: Notification = {
  id: '',
  message: ''
}

export default function notifications (state: Notification = initialState, action: any) {
  switch (action.type) {
    case DISPLAY_NOTIFICATION:
      return {
        id: Math.random().toString(36).slice(2),
        message: action.message
      }

    default:
      return state
  }
}
