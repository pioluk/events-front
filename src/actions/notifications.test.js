// global describe,test,expect

import * as actions from './notifications'

describe('notifications actions', () => {
  it('it should create action to display a notifications', () => {
    const message = 'Successfully logged in.'
    expect(actions.displayNotification(message)).toMatchSnapshot()
  })
})
