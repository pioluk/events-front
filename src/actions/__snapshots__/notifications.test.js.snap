exports[`notifications actions it should create action to display a notifications 1`] = `
Object {
  "message": "Successfully logged in.",
  "type": "DISPLAY_NOTIFICATION",
}
`;
